app/models/flight.rb

```diff
@@ -2,11 +2,33 @@ class Flight < ApplicationRecord
   belongs_to :airplane
   belongs_to :departure_airport, class_name: 'Airport'
   belongs_to :arrival_airport, class_name: 'Airport'
+  belongs_to :departure_city, class_name: 'City'
+  belongs_to :arrival_city, class_name: 'City'
   has_many :itinerary_flights
   has_many :itineraries, through: :itinerary_flights
 
   validates :number, presence: true, uniqueness: true
   validates :departure_date, presence: true
   validates :arrival_date, presence: true
   validates :price, presence: true
+
+  scope :availables, -> { where('available = ? AND departure_date >= ?', true, Date.today + 4.hours) }
```

Comment:
also a nitpick. It's better to read this as singular in english
```suggestion
  scope :available, -> { where('available = ? AND departure_date >= ?', true, Date.today + 4.hours) }
```

Reply:
done!